### DISCUSSION TITLE
Socket.io won't connect on client side [SSR]

### DISCUSSION THREAD

**Discussion by DrakoPOD** [quasarframework/quasar#10520] [CLI - SSR mode]
I trying to run socket.io for a communication between my server and my client for data visualization in real time. It is assumed that when I call the function `io()` it connects to the server, without specify the host, In the console show the petition from socket.io:
`[Quasar Dev Webserver] /socket.io/?EIO=4&transport=polling&t=NkJn1C1 -> request took: 9ms`
But nothing happens. Socket.io should use the same host that the quasar app. I have configured the socket server in a middleware.

Another problem, when I load the app in dev mode and change a file, the app restart the server, but always occur an error `port 3000 is in use`, apparently Quasar doesn't close the Socket.io port before restart.
```js
import { ssrMiddleware } from 'quasar/wrappers';
import express from 'express';
import http from 'http';
import { Server } from 'socket.io';

const port = process.env.SOCKET_PORT || 3000

// more info on params: https://v2.quasar.dev/quasar-cli/developing-ssr/ssr-middlewares
export default ssrMiddleware(({ app /*, resolveUrlPath, publicPath, render */ }) => {

  const server = http.createServer(app);
  const io = new Server(server);

  io.on('connection', (socket) => {
    console.log('new client arrive');
    socket.on('disconnect', () => {
      console.log('Client disconnected');
    })

    socket.onAny((eventName, ...args) => {
      console.log(eventName, args);
    })

  });

  server.listen(port, () => {
    console.log(`Server on port: ${port}`);
  });
});
```

**Comment by ontwikkelfabriek** [comment]
@DrakoPOD 

`ssrMiddleware ` runs on the server not the client.

The Sockets.io on the client knows nothing about quasar ssrMiddleware

**Reply by DrakoPOD** [reply]
Yes, but this is not my problem, in Socket.io docs, sais that Socket.io  it defaults to trying to connect to the host that serves the page, as you see, I pass the Express instance.
```js
const server = http.createServer(app);
const io = new Server(server);
```
But I don't know is Quasar is redirecting the petition or something similar.

**Comment by ontwikkelfabriek** [comment]
@DrakoPOD 

I have no reason to believe when looking at the Quasar middleware docs that sockets(.io) are supported in middleware. Just the included express server.

```javascript
const server = http.createServer(app); 
const io = new Server(server);
```

This code runs on the server part, the client knows nothing about this. 

 But I am no expert on middleware/SSR ( it's pretty new to Quasar). For better support I would ask around in the Quasar discord.

**Reply by DrakoPOD** [reply]
I think I know what is my problem, is because I'm using app express instance, but I need the server express instance generated by `app.listen`

Do you know how to get it? I can't find it
